diff -aur StreamDVD-0.4/mplex/inputstrm.cpp.old StreamDVD-0.4/mplex/inputstrm.cpp
--- StreamDVD-0.4/mplex/inputstrm.cpp.old	2004-03-08 23:10:53.000000000 +0100
+++ StreamDVD-0.4/mplex/inputstrm.cpp	2008-07-07 23:28:19.000000000 +0200
@@ -23,6 +23,7 @@
 
 #include "config.h"
 #include <assert.h>
+#include <climits>
 
 #include "mjpeg_types.h"
 #include "fastintfns.h"
diff -aur StreamDVD-0.4/mplex/multiplexor.cpp.old StreamDVD-0.4/mplex/multiplexor.cpp
--- StreamDVD-0.4/mplex/multiplexor.cpp.old	2004-03-08 23:10:53.000000000 +0100
+++ StreamDVD-0.4/mplex/multiplexor.cpp	2008-07-07 23:28:19.000000000 +0200
@@ -22,6 +22,7 @@
 #include "config.h"
 #include <math.h>
 #include <stdlib.h>
+#include <cstring>
 
 #include "mjpeg_types.h"
 #include "mjpeg_logging.h"
diff -aur StreamDVD-0.4/mplex/padstrm.cpp.old StreamDVD-0.4/mplex/padstrm.cpp
--- StreamDVD-0.4/mplex/padstrm.cpp.old	2004-03-08 23:10:53.000000000 +0100
+++ StreamDVD-0.4/mplex/padstrm.cpp	2008-07-07 23:28:19.000000000 +0200
@@ -25,6 +25,7 @@
 #endif
 
 #include "padstrm.hpp"
+#include <cstring>
 
 
 
diff -aur StreamDVD-0.4/mplex/zalphastrm_in.cpp.old StreamDVD-0.4/mplex/zalphastrm_in.cpp
--- StreamDVD-0.4/mplex/zalphastrm_in.cpp.old	2004-03-08 23:10:53.000000000 +0100
+++ StreamDVD-0.4/mplex/zalphastrm_in.cpp	2008-07-07 23:28:19.000000000 +0200
@@ -23,6 +23,7 @@
 #include "config.h"
 #include <math.h>
 #include <stdlib.h>
+#include <cstring>
 
 #include "zalphastrm.hpp"
 #include "interact.hpp"
